global_helpers/panther_greynoise_helpers.py

```diff
@@ -0,0 +1,155 @@
+import ast
+import datetime
+
+from dateutil import parser
+from panther_base_helpers import PantherUnexpectedAlert, deep_get
+
+
+class GreyNoiseBasic:
+    def __init__(self, event):
+        self.noise = deep_get(event, "p_enrichment", "greynoise_noise_basic")
+        if self.noise is None:
+            self.advanced_enabled = deep_get(event, "p_enrichment", "greynoise_noise_advanced")
+            if self.advanced_enabled is not None:
+                raise PantherUnexpectedAlert(
+                    "This account is configured with an Advanced GreyNoise Subscription.\
+                     Please use GreyNoiseAdvanced"
+                )
+
+    def ip_address(self, match_field) -> str:
+        return deep_get(self.noise, match_field, "ip")
+
+    def classification(self, match_field) -> str:
+        return deep_get(self.noise, match_field, "classification")
+
+    def actor(self, match_field) -> str:
+        return deep_get(self.noise, match_field, "actor")
+
+    def url(self, match_field) -> str:
+        return f"www.greynoise.io/viz/ip/{deep_get(self.noise, match_field, 'ip')}"
+
+
+class GreyNoiseAdvanced:
+    def __init__(self, event):
+        self.noise = deep_get(event, "p_enrichment", "greynoise_noise_advanced")
+        if self.noise is None:
+            self.basic_enabled = deep_get(event, "p_enrichment", "greynoise_noise_basic")
+            if self.basic_enabled is not None:
+                raise PantherUnexpectedAlert(
+                    "This account is configured with an Advanced GreyNoise Subscription.\
+                     Please use GreyNoiseAdvanced"
+                )
```

Comment:
is this right? should we check if advanced is set before saying this? .. same applies to all the other exceptions

Reply:
So, this returning None is ok if there were no fields matched by GreyNoise which will happen pretty frequently. The condition we are concerned about is someone looking for basic when they have advanced in the event. That would indicate the use of the incorrect helper